app/server/src/static-site-builder/wodinBuilder.ts

```diff
@@ -0,0 +1,97 @@
+import path from "path";
+import fs from "fs";
+import { ConfigReader } from "../configReader";
+import { AppConfig, WodinConfig } from "../types";
+import { version as wodinVersion } from "../version";
+import { render } from "mustache";
+import { ConfigController } from "../controllers/configController";
+import { AppFileReader } from "../appFileReader";
+import axios from "axios";
+import { processArgs } from "./args";
+
+const mkdirForce = (path: string) => {
+    if (!fs.existsSync(path)) {
+        fs.mkdirSync(path);
+    }
+};
+
+const { configPath, destPath, viewsPath } = processArgs();
+
+const pathResolved = path.resolve(configPath);
+const configReader = new ConfigReader(pathResolved);
+const defaultCodeReader = new AppFileReader(`${pathResolved}/defaultCode`, "R");
+const appHelpReader = new AppFileReader(`${pathResolved}/help`, "md");
+const wodinConfig = configReader.readConfigFile("wodin.config.json") as WodinConfig;
+
+mkdirForce(destPath);
+const appsPathFull = path.resolve(destPath, wodinConfig.appsPath);
+if (fs.existsSync(appsPathFull)) {
+    fs.rmSync(appsPathFull, { recursive: true });
+}
+fs.mkdirSync(appsPathFull);
+
+const sessionId = null;
+const shareNotFound = null;
+const baseUrl = wodinConfig.baseUrl.replace(/\/$/, "");
+
+const appNames = fs.readdirSync(path.resolve(configPath, wodinConfig.appsPath)).map(fileName => {
+    return /(.*)\.config\.json/.exec(fileName)![1];
+});
+
+appNames.forEach(async appName => {
+    const appNamePath = path.resolve(appsPathFull, appName);
+    fs.mkdirSync(appNamePath);
+
+    const configWithDefaults = ConfigController.readAppConfigFile(
+        appName, wodinConfig.appsPath, baseUrl, configReader, defaultCodeReader, appHelpReader
+    );
+    const readOnlyConfigWithDefaults = { ...configWithDefaults, readOnlyCode: true };
+    const configResponse = {
+        status: "success",
+        errors: null,
+        data: readOnlyConfigWithDefaults
+    };
+    fs.writeFileSync(path.resolve(appNamePath, "config.json"), JSON.stringify(configResponse));
+    
+
+    const versionsResponse = await axios.get("http://localhost:8001/");
+    fs.writeFileSync(path.resolve(appNamePath, "versions.json"), JSON.stringify(versionsResponse.data));
+
+
+    const runnerResponse = await axios.get("http://localhost:8001/support/runner-ode");
+    fs.writeFileSync(path.resolve(appNamePath, "runnerOde.json"), JSON.stringify(runnerResponse.data));
+
+    if (configWithDefaults.appType === "stochastic") {
+        const runnerResponse = await axios.get("http://localhost:8001/support/runner-discrete");
+        fs.writeFileSync(path.resolve(appNamePath, "runnerDiscrete.json"), JSON.stringify(runnerResponse.data));
+    }
```

Comment:
they are! the reason i did it per app is the request fired is relative to the app, so like apps/day1/runner/ode or something like that, so it was easier to have a duplicate in each app, obviously this doesnt scale well but i dont think people ever have more than 5-6 apps, so that much duplication of that file seemed alright for slightly simpler code but happy to change it if needed